lib/spack/spack/build_systems/autotools.py

```diff
@@ -110,70 +109,49 @@ def _do_patch_config_files(self):
                     continue
                 except Exception as e:
+                    my_config_files.append(conf)
                     tty.debug(e)
-            else:
+            if not my_config_files:
                 continue
 
-            # Look for a spack-installed automake package
+            system_config_root = []
             if 'automake' in self.spec:
-                automake_dir = 'automake-' + str(self.spec['automake'].version)
-                automake_path = os.path.join(self.spec['automake'].prefix,
-                                             'share', automake_dir)
-                path = os.path.join(automake_path, config_file)
-                if os.path.exists(path):
-                    config_files[config_name] = path
-            # Look for the system's config.guess
-            if (config_files[config_name] is None and
-                    os.path.exists('/usr/share')):
-                automake_dir = [s for s in os.listdir('/usr/share') if
-                                "automake" in s]
-                if automake_dir:
-                    automake_path = os.path.join('/usr/share', automake_dir[0])
-                    path = os.path.join(automake_path, config_file)
-                    if os.path.exists(path):
-                        config_files[config_name] = path
-            if config_files[config_name] is not None:
-                try:
-                    config_path = config_files[config_name]
-                    my_config_path = my_config_files[config_name]
-
-                    check_call([config_path] + config_args[config_name],
-                               stdout=PIPE, stderr=PIPE)
-
-                    m = os.stat(my_config_path).st_mode & 0o777 | stat.S_IWUSR
-                    os.chmod(my_config_path, m)
-                    shutil.copyfile(config_path, my_config_path)
-                    continue
-                except Exception as e:
-                    tty.debug(e)
-
-            raise RuntimeError('Failed to find suitable ' + config_file)
+                system_config_root.append(self.spec['automake'].prefix)
+            system_config_root.append('/usr/share')
+            config_path = None
+            for system_root in system_config_root:
+                for path in find(system_root, config_file):
+                    try:
+                        check_call([path] + config_args[config_name],
+                                   stdout=PIPE, stderr=PIPE)
+                        config_path = path
+                        break
+                    except Exception as e:
+                        tty.debug(e)
+                        continue
+
+            if config_path is not None:
+                for myconf in my_config_files:
+                    copy(config_path, myconf)
+                break
+            else:
+                raise RuntimeError('Failed to find suitable ' + config_file)
```

Comment:
This block mainly checks which script should be used to update `config.guess` and `config.sub`. Maybe it's also worth being extracted in its own function, be it nested here or a private method on its own.